src/Compiler.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+/**
+ * This file is part of the backendtea/php-optimizer package.
+ * Copyright (c) 2018 Gert de Pagter
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace PhpOptimizer;
+
+use Seld\PharUtils\Timestamps;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Compiler class creates a PHAR file from this package. Heavily inspired by Composer's Compiler class.
+ *
+ * @author Loek van der Linde <lind0077@hz.nl>
+ */
+class Compiler
+{
+    /**
+     * @var string
+     */
+    private $versionDate;
+
+    public function compile(string $filename = 'php-optimize.phar'): void
+    {
+        // TODO figure out what php compatibility should be, now it's 7+ due to typehint
+
+        if (\file_exists($filename)) {
+            \unlink($filename);
+        }
+
+        $date = new \DateTime('now', new \DateTimeZone('UTC'));
+        $this->versionDate = $date->format('c');
+
+        $phar = new \Phar($filename, 0, 'php-optimize.phar');
+        $phar->setSignatureAlgorithm(\Phar::SHA1);
+
+        // PHAR is a stream, so open it
+        $phar->startBuffering();
+
+        $finder = new Finder();
+        $finder
+            ->files()
+            ->ignoreVCS(true)
+            ->name('*.php')
+            ->notName('Compiler.php')
+            ->in(__DIR__)
+        ;
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file);
+        }
+
+        $finder = new Finder();
+        $finder->files()
+            ->ignoreVCS(true)
+            ->name('*.php')
+            ->name('LICENSE')
+            ->exclude('Tests')
+            ->exclude('tests')
+            ->exclude('docs')
+            ->in(__DIR__ . '/../vendor/nikic/php-parser/')
```

Comment:
Could we jut add  `../vendor`, if we run install the dependencies with not dev?

Reply:
On that note, if possible, could you add a make target that runs these steps?

Reply:
That requires some heavy tinkering, since I've put all the necessary libraries for PHAR creation in the --dev corner. I'll take a look at it, but for now I'm skipping it.